packages/js/product-editor/src/blocks/product-fields/inventory-email/edit.tsx

```diff
@@ -44,6 +44,8 @@ export function Edit( {
 
 	const id = useInstanceId( BaseControl, 'low_stock_amount' ) as string;
 
+	const errorContext = 'inventory';
```

Comment:
I don't recommend hard coding this, and same counts for the other changes.
Because these blocks could be used in other templates, what if a consumer of that template decides to name that tab something different than `inventory`? Than this won't work anymore.

Have you seen the `BlockFill` implementation: https://github.com/woocommerce/woocommerce/blob/tweak/number-control/packages/js/product-editor/src/components/block-slot-fill/block-fill.tsx#L23-L40

It makes use of the block client ID, we than can make use of the `block-editor` data store and get all the parents of a certain type: `getBlockParentsByBlockName`, so in theory we could get all the block tabs for that specific block. Than we know exactly which tab we should go to, no matter where the user inserted it in the template.
See additional supported functions here: https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/data/data-core-block-editor.md

Something like this would also be good to get some team feedback on earlier, just throwing it out in the public channel if this is the best option.

Reply:
I kinda like the idea of having some sort of functionality that can tell the block about its parent tab, but my main concern here is performance. Should every single block that register a validator have to search for its top most parent which name matches the `woocommerce/product-tab`? Can we have a way to tell such thing to the block during the template registration/definition/creation instead?

Reply:
Yeah I don't think this should be done on block creation/rendering. Only when the error happens with providing the `clientId`. Sounds like this may already be something you guys considered.